contracts/extensions/VotingToken.sol

```diff
@@ -0,0 +1,133 @@
+  function identifier() public override pure returns (bytes32) {
+    return keccak256("VotingToken");
+  }
+
+  /// @notice Return the version number
+  /// @return The version number
+  function version() public pure override returns (uint256) {
+    return 1;
+  }
+
+  // [motionId][user] => tokenBalance
+  mapping (uint256 => mapping (address => uint256[])) influences;
+
+  // [motionId] => lockId
+  mapping (uint256 => uint256) lockIds;
+
+  // Public
+
+  /// @notice Set influence for a motion
+  /// @param _motionId The id of the motion
+  function setInfluence(uint256 _motionId) public motionExists(_motionId) {
+    if (influences[_motionId][msg.sender].length == 0) {
+      uint256 balance = tokenLocking.getUserLock(token, msg.sender).balance;
+      influences[_motionId][msg.sender] = new uint256[](NUM_INFLUENCES);
+      influences[_motionId][msg.sender][0] = balance;
+    }
+  }
+
+  /// @notice Get the user influence in the motion
+  /// @param _motionId The id of the motion
+  /// @param _user The user in question
+  function getInfluence(uint256 _motionId, address _user)
+    public
+    view
+    override
+    returns (uint256[] memory influence)
+  {
+    influence = influences[_motionId][_user];
+  }
+
+  function postReveal(uint256 _motionId, address _user) internal override {
+    if (lockIds[_motionId] == 0) {
+      // This is the first reveal that has taken place in this motion.
+      // We lock the token for everyone to avoid double-counting,
+      lockIds[_motionId] = colony.lockToken();
+    }
+
+    colony.unlockTokenForUser(_user, lockIds[_motionId]);
+  }
+
+  function postClaim(uint256 _motionId, address _user) internal override {
+    uint256 lockCount = tokenLocking.getUserLock(token, _user).lockCount;
+
+    // Lock may have already been released during reveal
+    if (lockCount < lockIds[_motionId]) {
+      colony.unlockTokenForUser(_user, lockIds[_motionId]);
+    }
+  }
+
+  /// @notice Create a motion in the root domain
+  /// @param _altTarget The contract to which we send the action (0x0 for the colony)
+  /// @param _action A bytes array encoding a function call
+  function createMotion(address _altTarget, bytes memory _action)
+    public
+    notDeprecated
+  {
+    createMotionInternal(1, UINT256_MAX, _altTarget, _action, NUM_INFLUENCES);
+    motions[motionCount].maxVotes[0] = ERC20Extended(token).totalSupply();
+  }
+
+  /// @notice Stake on a motion
+  /// @param _motionId The id of the motion
+  /// @param _permissionDomainId The domain where the extension has the arbitration permission
+  /// @param _childSkillIndex For the domain in which the motion is occurring
+  /// @param _vote The side being supported (0 = NAY, 1 = YAY)
+  /// @param _amount The amount of tokens being staked
+  function stakeMotion(
+    uint256 _motionId,
+    uint256 _permissionDomainId,
+    uint256 _childSkillIndex,
+    uint256 _vote,
+    uint256 _amount
+  )
+    public
+  {
+    setInfluence(_motionId);
```

Comment:
Hmm, good point. I suppose setting influence at `revealVote` makes the most sense.

Reply:
If we are in favor of removing restrictions on staking as a function of influence, then we can move all influence setting to `revealVote`, which would let reputation, token, hybrid voting all play nicely together (since that is when we would lock the token). I don't see a real problem with removing the restriction on staking as a function of rep -- sure, someone with little rep could now fully stake a motion, but what of it? I guess the risk is someone can pay-to-play vis a vis putting motions on the colony...

Reply:
An alternative would be to require a reputation proof for staking (without saving the influence). Then we can save the influence during the reveal phase. Although then I wonder if we need to save the influence at all?

Reply:
Although no, we need to record the influence during the reveal phase to see if we can "accelerate" the timeline by hitting some vote requirement (e.g. 80%).

Hmm... I suppose we can dispense with saving influence entirely and just require reputation proofs / check token balances each time? This could allow some token balance manipulation in the submit phase (in order to artificially hit the submission requirement) but wouldn't allow any double-counting in the reveal phase, which arguably is more important?

Reply:
After some conversation with Product, this was resolved by removing the fast-forward functionality for token & hybrid voting and checking influence during each transaction (rather than putting it in storage). This allows us to leave tokens unlocked till reveal while preventing any manipulation.